src/net.cpp

```diff
@@ -1097,10 +1097,13 @@ void ThreadDNSAddressSeed()
         } else {
             vector<CNetAddr> vIPs;
             vector<CAddress> vAdd;
-            if (LookupHost(seed.host.c_str(), vIPs))
+            if (LookupHost(((seed.fXoredIPs ? "x." : "") + seed.host).c_str(), vIPs))
             {
                 BOOST_FOREACH(CNetAddr& ip, vIPs)
                 {
+                    if (seed.fXoredIPs)
+                        ip.XORAddress();
```

Comment:
Be really careful with references (CNetAddr& ip) and this function mutating its state in-place, as it could garble existing data structures. In this case it doesn't give problems as the vector being iterated over is being thrown away after use. But from a peace of mind point of view I'd prefer a function `ip.XORedAddress() const` that returns a new address instead of overwrites itself.